File: Models/FileModel.cs
using System.Collections.Generic;
using System;

namespace Cont5.Models.File {
    public class GenerateDownloadRequest {
        public string Path { get; set; }
        public GenerateDownloadRequest(string path){
            this.Path = path;
        }
    }
    public class GenerateDownloadModel {
        public string Link { get; set; }
        public DateTime GeneratedDt { get; set; }
        public DateTime ExpiryDt { get; set; }
    }
    public class GetMovablePathModel {
        public string Name { get; set; }
        public string Path { get; set; }
        public GetMovablePathModel (string Name, string Path) {
            this.Name = Name;
            this.Path = Path;
        }
    }
    public class SanitizedHiddenPathModel{
        public bool IsPathChanged { get; set; }
        public string Path { get; set; }
    }
    public class DeleteFileRequest { 
        public string Path { get; set; }
    }

    public class DeleteFileModel {
        public bool Result { get; set; }
    }
    public class CreateNewFolderRequest { 
        public string Path { get; set; }
        public string Name { get; set; }
    }

    public class CreateNewFolderModel {
        public bool Result { get; set; }
    }
    public class GetJustMovedModel {
        public List<File> FileList { get; set; }
        public class File {
            public string Name { get; set; }
            public string Path { get; set; }
        }
    }
    public class SubmitMoveFileRequest { 
        public string SourceFile { get; set; }
        public string TargetPath { get; set; }
    }

    public class SubmitMoveFileModel {
        public bool Result { get; set; }
    }
    public class SubmitCreateMoveFileRequest { 
        public string SourceFile { get; set; }
        public string TargetPath { get; set; }
    }

    public class SubmitCreateMoveFileModel {
        public bool Result { get; set; }
    }
    public class DownloadFileRequest {
        public string Path { get; set; }
        public string Token { get; set; }
    }
    public class DownloadFileModel {
        public SanitizePathModel SanitizedPath { get; set; }
    }
    public class ScanPathRequest {
        public string Path { get; set; }
    }
    public class ScanPathModel {
        public bool IsDirectory { get; set; }
        public string CurrentPath { get; set; }
        public bool ShowFolderOption { get; set; }
        public bool IsDeletable { get; set; }
        public bool IsBookmarkable { get; set; }
        public bool IsFolderCreatable { get; set; }
        public bool isMovable { get; set; }
        public List<GetMovablePathModel> MovableFolderPath { get; set; }
        public List<ContFolder> FolderList { get; set; }
        public List<ContFile> FileList { get; set; }
        public string WindowsPlayer { get; set; }
        public string WindowsUrlScheme { get; set; }
    }

    public class SubmitWatchedRequest { 
        public string Path { get; set; }
    }

    public class SubmitWatchedModel {
        public bool Result { get; set; }
    }

    public class SubmitNoteRequest { 
        public string FileName { get; set; }
        public string Note { get; set; }
    }

    public class SubmitNoteModel {
        public bool Result { get; set; }
    }

    public class SanitizePathModel {
        public string Path { get; set; }
        public string Url { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string Extension { get; set; }
        public string RelativePath { get; set; }
        public bool IsDirectory { get; set; }
        public bool IsPathChanged { get; set; }
        public bool IsVideo { get; set; }
        public bool IsInAnimeDirectory { get; set; }
        public bool IsInAnimeEnd { get; set; }
    }

    public class ContFolder {
        public string Name { get; set; }
        public bool IsNew { get; set; }
    }

    public class ContFile {
        public string Name { get; set; }
        public int Type { get; set; }
        public string Note { get; set; }
        public bool IsWatched { get; set; }
        public bool IsNew { get; set; }
        public string Link { get; set; }
        public string SubtitleUrl { get; set; }
        public string AnimeName { get; set; }
        public FileListPropModel fileListProp { get; set; }
    }
    
    public class FileListPropModel {
        public bool ShowDownloadButton { get; set; }
        public bool ShowDownloadSubtitleButton { get; set; }
        public bool ShowWatchedButton { get; set; }
    }

    public class GetSubtitleRequest {
        public string Path { get; set; }
    }

    public class GetSubtitleModel {
        public bool Result { get; set; }
        public string Url { get; set; }
    }

    public class UploadTextFileRequest {
        public string Path { get; set; }
        public string Content { get; set; }
    }

    public class UploadTextFileModel {
        public bool Result { get; set; }
    }
}

File: ClientApp/src/app/models/file.model.ts
import { SafeUrl, DomSanitizer } from "@angular/platform-browser";
import { Sanitizer } from "@angular/core";

export class ScanPathModel {
    public isDirectory: boolean;
    public currentPath: string;
    public folderList: ContFolder[];
    public fileList: ContFile[];
    public isDeletable: boolean;
    public isBookmarkable: boolean;
    public isMovable: boolean;
    public movableFolderPath: GetMovablePathModel[];
    public showFolderOption: boolean;
    public isFolderCreatable: boolean;
    public windowsPlayer: string;
    public windowsUrlScheme: string;
}
export class ContFolder {
    public name: string;
    public isNew: boolean;
}
export class ContFile {
    public name: string;
    public type: number;
    public note: string;
    public isWatched: boolean;
    public isNew: boolean;
    public link: string;
    public subtitleUrl: string;
    public animeName: string;
    public fileListProp: FileListPropModel;
}
export class FileListPropModel {
    public showDownloadButton: boolean;
    public showWatchedButton: boolean;
    public showDownloadSubtitleButton: boolean;
}
export class GetMovablePathModel {
    public name: string;
    public path: string;
}
export class GenerateDownloadModel {
    public link: string;
    public generatedDt: Date;
    public expiryDt: Date;
}
export class DeleteFileModel {
    public result: boolean;
}
export class SubmitNoteModel {
    public result: boolean;
}
export class SubmitMoveModel {
    public result: boolean;
}
export class SubmitCreateMoveModel {
    public result: boolean;
}
export class GetSubtitleModel {
    public result: boolean;
    public url: string;
}
export class UploadTextFileModel {
    public result: boolean;
}
export class CustomPlayModel {
    public link: string;
    public mxLink: string;
    public windowsLink: string;
    public resumeLink: string;
    public windowsPlayer: string = "MPCBE";

    constructor(note: string, link: string, windowsPlayer: string, windowsUrlScheme: string) {
        this.link = link;
        this.mxLink = "intent:" + link + "#Intent;package=com.mxtech.videoplayer.pro;end";
        this.windowsLink = windowsUrlScheme + link;
        if (windowsPlayer) {
            this.windowsLink += ":" + windowsPlayer;
            this.windowsPlayer = windowsPlayer;
        }
        if (note) {
            let mxResumeTiming = this.getMxResumeTiming(note);
            if (mxResumeTiming) {
                this.resumeLink = "intent:"+link+"#Intent;package=com.mxtech.videoplayer.pro;i.position="+mxResumeTiming+";end";
            }
        }
    }

    getMxResumeTiming(note: string) {
        try {
            let output:string = "";
            let tempNote:string = note;
            let noteLength:number = tempNote.length;
            
            let hrs:number = 0, mins:number = 0, secs:number = 0;
            if (noteLength === 4) {
                mins = parseInt(tempNote.substr(0, 2));
                secs = parseInt(tempNote.substr(2, 2));
                output = ((mins * 60000) + (secs * 1000)).toString();
            } else if (noteLength === 6) {
                hrs = parseInt(tempNote.substr(0, 2));
                mins = parseInt(tempNote.substr(2, 2));
                secs = parseInt(tempNote.substr(4, 2));
                output = ((hrs * 3600000) + (mins * 60000) + (secs * 1000)).toString();
            } else {
                return false;
            }
            return output;
        } catch (Exception) {
            return null;
        }
    }

    public reset() {
        this.mxLink = null;
        this.resumeLink = null;
    }
}